functions.py

```diff
@@ -26,12 +34,223 @@ def getServiceFromUser():
+    pw = hashlib.sha512(pw.encode('utf-8')).hexdigest()
+    if addUser(username, pw):
+        return True
+    else:
+        print("Sorry, that username is already taken")
+        signUpUser()
+
+def handleLogin():
+    print("Do you want to log in or start a new account?\n(Enter the number of your choice)\n")
+    print("(1) Log In")
+    print("(2) Start New Account\n")
+    option = getUserInput("> ")
+    if option == "1":
+        loginUser()
+
+    elif option == "2":
+        signUpUser()
+
+    else:
+        print("Please enter a valid option\n")
+        handleLogin()
+
+def welcomeMessage():
+    print("\n\n")
+    print("##################################################")
+    print("# Welcome to Passman!")
+    print("##################################################\n\n")
+
+def generatePasswordPrompt():
+    #TODO
+    print("todo")
+
+def listServicesPrompt():
+    for service in getAllServices():
+        if service['serviceUserName'] == "":
+            if service['serviceUrl'] == "":
+                print(service['service'])
+            else:
+                print(service['service'],service['serviceUrl'],sep='\t')
+        else:
+            if service['serviceUrl'] == "":
+                print(service['service'],"-",service['serviceUserName'],sep='\t')
+            else:
+                print(service['service'],service['serviceUserName'],service['serviceUrl'],sep='\t')
+    return True
+def addServicePrompt(name="",usname="",url=""):
+    if not usname == "":
+        if checkIfServiceExists(name):
+            print("Service already exists.")
+            return False
+    elif not name=="":
+        if checkIfServiceExists(name):
+            print("Service already exists.")
+            return False
+        usname = getUserInput("Username: ")
+    else:
+        name=getUserInput("Entry name: ")
+        if checkIfServiceExists(name):
+            print("Service already exists.")
+            return False
+        else:
+            usname = getUserInput("Enter username: ")
+
+    password = getUserInput("Password [leave blank to generate]: ", True)
+    if password == "":
+        generatePasswordPrompt()
+    url=getUserInput("Service URL: ")
+
+    return addService(name, password, url, usname)
+def removeServicePrompt(sname=""):
+    if sname=="":
+        sname = getUserInput("Enter service to be deleted: ")
+        while not checkIfServiceExists(sname):
+            print("Service not found.")
+            sname = getUserInput("Enter service to be deleted: ")
+    else:
+        if not checkIfServiceExists(sname):
+            print("Service not found.")
+            return False
+    service = getServiceByName(sname)
+    print("Delete ",service['service'],". This cannot be undone.",sep="")
+    confirm = getUserInput("Are you sure? (y/N)")
+    if confirm == "y" or confirm =="Y":
+        servname = service['service']
+        success = removeService(service)
```

Comment:
@nosnixster This was the bug you texted me about. removeService takes a name only. 

Your pull from the db was getting called, but it was trying to match a dictionary and a name, and failing.

Fixed here: 757d833